### PR TITLE
Security: Update ActionView to 5.2.4.2

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
https://github.com/advisories/GHSA-65cv-r6x7-79hv (CVE-2020-5267).

We don’t use the affected helpers (j, escape_javascript) in the app, but
putting this in for future safety.

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
Looking at the failing Travis builds, we're getting `Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true` when calling, I think, `ActiveStorage::Blob#service_url`.

Looking at `activestorage/test/service/disk_service_test.rb` the [diff in `rails/rails` for this upgrade](https://github.com/rails/rails/compare/v5.2.2.1...v5.2.4.2) suggests that something has changed in Rails. If I get more time before coming off support, I'll look into it.

**Comment by lozette** [comment]
I've done another commit to switch from using `service_url` to `rails_blob_url` but I just don't know if it's correct?? I followed this guidance: https://github.com/rails/rails/issues/32500

Wouldn't mind someone who might know what they're doing to take a look 😬

**Comment by leeky** [comment]
@lozette Hmmm.. `rails_blob_url` generates a signed URL which redirects to the S3 one. I have a suspicion that this didn't work properly when we tried using it before. Might be because ingest doesn't follow redirects, or they're one time URLs, which would break ActiveJob retries.

Sorry to be vague, it's been a while since I looked at the S3 part of the system :-/

File: spec/support/active_storage.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 RSpec.configure do |config|
   config.before(:each) do
-    ActiveStorage::Current.host = 's3.example.com'
+    Rails.application.routes.default_url_options[:host] = 's3.example.com'
```
Comment (leeky) [inline spec/support/active_storage.rb]: This sets the default host to `s3.example.com` everywhere - so, for example, if we sent an email in a test  links would contain `s3.example.com` as the host, which is not correct.

`ActiveStorage::Current.host` sets the host for S3 only, but for some reason this isn't working after Rails v5.2.4.

**Comment by leeky** [comment]
@lozette The good news is that the ingest process grabs files from S3 using the `aws-sdk-s3` library instead of going through ActiveStorage. So I say we merge this, and ask Timur to test it on staging - he should check that the admin and user-facing sites can both download a submission (XLS) file and we should be good to deploy this 🎉

**Comment by lozette** [comment]
Thank you @leeky!! Really appreciate you taking the time

Can you approve it & I'll merge

**Comment by leeky** [comment]
Closing this in favour of https://github.com/dxw/DataSubmissionServiceAPI/pull/607

**Pull request closed without merging**